Examples/Examples/All Examples/SceneKitExample.swift

```diff
@@ -0,0 +1,228 @@
+        pointNode.light?.intensity = 3000
+        pointNode.position = SCNVector3Make(0, 25, 0)
+        modelNode.addChildNode(pointNode)
+    }
+
+    internal func makeTranslationMatrix(tx: Double, ty: Double, tz: Double) -> simd_double4x4 {
+        var matrix = matrix_identity_double4x4
+
+        matrix[3, 0] = tx
+        matrix[3, 1] = ty
+        matrix[3, 2] = tz
+
+        return matrix
+    }
+
+    internal func makeScaleMatrix(xScale: Double, yScale: Double, zScale: Double) -> simd_double4x4 {
+        var matrix = matrix_identity_double4x4
+
+        matrix[0, 0] = xScale
+        matrix[1, 1] = yScale
+        matrix[2, 2] = zScale
+
+        return matrix
+    }
+
+    public func render(_ parameters: CustomLayerRenderParameters, mtlCommandBuffer: MTLCommandBuffer, mtlRenderPassDescriptor: MTLRenderPassDescriptor) {
+        let m = parameters.projectionMatrix
+
+        // It is essential to use double precision for computation below: using simd instead
+        // of SceneKit matrix operations.
+        var transformSimd = matrix_identity_double4x4
+        transformSimd[0, 0] = m[0].doubleValue
+        transformSimd[0, 1] = m[1].doubleValue
+        transformSimd[0, 2] = m[2].doubleValue
+        transformSimd[0, 3] = m[3].doubleValue
+        transformSimd[1, 0] = m[4].doubleValue
+        transformSimd[1, 1] = m[5].doubleValue
+        transformSimd[1, 2] = m[6].doubleValue
+        transformSimd[1, 3] = m[7].doubleValue
+        transformSimd[2, 0] = m[8].doubleValue
+        transformSimd[2, 1] = m[9].doubleValue
+        transformSimd[2, 2] = m[10].doubleValue
+        transformSimd[2, 3] = m[11].doubleValue
+        transformSimd[3, 0] = m[12].doubleValue
+        transformSimd[3, 1] = m[13].doubleValue
+        transformSimd[3, 2] = m[14].doubleValue
+        transformSimd[3, 3] = m[15].doubleValue
+
+        // Model is using metric unit system: scale x and y from meters to mercator and keep z is in meters.
+        let meterInMercatorCoordinateUnits = try! 1.0 / (Projection.getMetersPerPixelAtLatitude(forLatitude: modelOrigin.latitude, zoom: parameters.zoom))
+        let modelScale = makeScaleMatrix(xScale: meterInMercatorCoordinateUnits, yScale: -meterInMercatorCoordinateUnits, zScale: 1)
+
+        // Translate scaled model to model origin (in web mercator coordinates) and elevate to model origin's altitude (in meters).
+        let origin = try! Projection.project(for: modelOrigin, zoomScale: pow(2, parameters.zoom))
+        var elevation = 0.0
+        if let elevationData = parameters.elevationData, let elevationValue = elevationData.getElevationFor(self.modelOrigin) {
+            elevation = elevationValue.doubleValue
+        }
+        let translateModel = makeTranslationMatrix(tx: origin.x, ty: origin.y, tz: elevation)
+
+        let transform = transformSimd * translateModel * modelScale
+
+        let scnMat = SCNMatrix4(
+            m11: Float(transform[0, 0]),
```

Comment:
Have you noticed any jittering of the model relative to the world?

Reply:
Now that simd is used, no.